modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -182,7 +181,8 @@ private List<String> findRoutingPaths(String indexName, Settings allSettings, Li
                         // Since FieldMapper.parse modifies the Map passed in (removing entries for "type"), that means
                         // that only the first pathMatch passed in gets recognized as a time_series_dimension. To counteract
                         // that, we wrap the mappingSnippet in a new HashMap for each pathMatch instance.
```

Comment:
Remove the last sentence, and replace it with "To avoid this, each parsing call uses a different mapping snippet"